resultstore-search-client/src/components/SearchBar/index.tsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import { createStyles, Theme, makeStyles } from '@material-ui/core/styles';
+import OutlinedInput from '@material-ui/core/OutlinedInput';
+import InputAdornment from '@material-ui/core/InputAdornment';
+import FormHelperText from '@material-ui/core/FormHelperText';
+import Search from '@material-ui/icons/Search';
+import FormControl from '@material-ui/core/FormControl';
+import { searchInvocations } from '../../api/client/index';
+import * as invocation_pb from '../../api/invocation_pb';
+
+const searchPlaceholder = 'Search Invocations';
+
+const useStyles = makeStyles((theme: Theme) =>
+    createStyles({
+        margin: {
+            margin: theme.spacing(2),
+        },
+        root: {
+            display: 'flex',
+            flexWrap: 'wrap',
+        },
+    })
+);
+
+interface State {
+    query: string;
+    invocations: Array<invocation_pb.Invocation>;
+    errorText: string;
+    hasError: boolean;
+}
+
+/*
+SearchBar component to search for invocations by query string
+*/
+const SearchBar = () => {
+    const [query, setQuery] = React.useState<State['query']>('');
+    const [, setInvocations] = React.useState<State['invocations']>([]);
```

Comment:
whoa i am old -- is this a new react feature? I can't tell how lifecycle works anymore

Reply:
This is react hooks https://reactjs.org/docs/hooks-intro.html.

Reply:
good job!

Reply:
Thanks!